Themis/ViewModels/Assessment/Programming/CodeEditorViewModel.swift

```diff
@@ -122,6 +122,8 @@ class CodeEditorViewModel: ExerciseRendererViewModel {
             }
             appendHighlight(feedbackId: feedbackId, range: range, path: file.path)
         }
+        
+        undoManager.endUndoGrouping() // undo group with addFeedback in AssessmentResult
```

Comment:
Wouldn't adding this to the appendHighlight function make more sense?

Reply:
`appendHighlight()` is also used when existing highlights are loaded from the server. Calling `.endUndoGrouping()` would cause a crash in this case since there would be no corresponding `.beginUndoGrouping()` calls